aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -521,30 +527,33 @@ impl AptosVM {
             is_account_init_for_sponsored_transaction(txn_data, self.features(), resolver)?;
 
         if is_account_init_for_sponsored_transaction {
-            let mut session = self.new_session(resolver, SessionId::run_on_abort(txn_data));
+            let mut abort_hook_session =
+                AbortHookSession::new(self, txn_data, resolver, prologue_change_set)?;
```

Comment:
I "think" we can kill this now with the respawn-able sessions?

Reply:
You mean we can implement deduction in the prologue?

Reply:
no, this abort session was a hack because we can't continue from previous session. the better behavior is to use a session to create account first instead of during abort time. (that's why I tried this before, but probably separate pr)